From 32b1ecff84fae3db66120880536f4a82c61af19e Mon Sep 17 00:00:00 2001
From: Daniel Kahn Gillmor <dkg@fifthhorseman.net>
Date: Tue, 16 Aug 2022 14:11:51 -0400
Subject: [PATCH] Drop bindgen layout tests
Forwarded: https://github.com/rusqlite/rusqlite/pull/1213

The auto-generated bindgen layout tests are architecture dependent and
cause breakage on 32-bit platforms at least.

Note that this does not yet remove the layout tests from
bindgen-bindings/bindgen_*.rs -- i'm not sure how those are generated.

Addresses: #722

diff --git a/build.rs b/build.rs
index e154a8d..d3ef8b2 100644
--- a/build.rs
+++ b/build.rs
@@ -607,6 +607,7 @@ mod bindings {
         }
 
         bindings
+            .layout_tests(false)
             .generate()
             .unwrap_or_else(|_| panic!("could not run bindgen on header {}", header))
             .write(Box::new(&mut output))
diff --git a/bindgen-bindings/bindgen_3.6.23.rs b/bindgen-bindings/bindgen_3.6.23.rs
index 673b11d..7910b58 100644
--- a/bindgen-bindings/bindgen_3.6.23.rs
+++ b/bindgen-bindings/bindgen_3.6.23.rs
@@ -349,21 +349,9 @@ pub struct sqlite3_file_sqlite3_io_methods {
                                                           ->
                                                               ::std::os::raw::c_int>,
 }
-#[test]
-fn bindgen_test_layout_sqlite3_file_sqlite3_io_methods() {
-    assert_eq!(::std::mem::size_of::<sqlite3_file_sqlite3_io_methods>() ,
-               104usize);
-    assert_eq!(::std::mem::align_of::<sqlite3_file_sqlite3_io_methods>() ,
-               8usize);
-}
 impl Clone for sqlite3_file_sqlite3_io_methods {
     fn clone(&self) -> Self { *self }
 }
-#[test]
-fn bindgen_test_layout_sqlite3_file() {
-    assert_eq!(::std::mem::size_of::<sqlite3_file>() , 8usize);
-    assert_eq!(::std::mem::align_of::<sqlite3_file>() , 8usize);
-}
 impl Clone for sqlite3_file {
     fn clone(&self) -> Self { *self }
 }
@@ -464,11 +452,6 @@ pub struct sqlite3_vfs {
                                                                       *mut ::std::os::raw::c_char)
                                                  -> ::std::os::raw::c_int>,
 }
-#[test]
-fn bindgen_test_layout_sqlite3_vfs() {
-    assert_eq!(::std::mem::size_of::<sqlite3_vfs>() , 136usize);
-    assert_eq!(::std::mem::align_of::<sqlite3_vfs>() , 8usize);
-}
 impl Clone for sqlite3_vfs {
     fn clone(&self) -> Self { *self }
 }
@@ -519,11 +502,6 @@ pub struct sqlite3_mem_methods {
                                                                   *mut ::std::os::raw::c_void)>,
     pub pAppData: *mut ::std::os::raw::c_void,
 }
-#[test]
-fn bindgen_test_layout_sqlite3_mem_methods() {
-    assert_eq!(::std::mem::size_of::<sqlite3_mem_methods>() , 64usize);
-    assert_eq!(::std::mem::align_of::<sqlite3_mem_methods>() , 8usize);
-}
 impl Clone for sqlite3_mem_methods {
     fn clone(&self) -> Self { *self }
 }
@@ -1386,11 +1364,6 @@ pub struct sqlite3_vtab {
     pub nRef: ::std::os::raw::c_int,
     pub zErrMsg: *mut ::std::os::raw::c_char,
 }
-#[test]
-fn bindgen_test_layout_sqlite3_vtab() {
-    assert_eq!(::std::mem::size_of::<sqlite3_vtab>() , 24usize);
-    assert_eq!(::std::mem::align_of::<sqlite3_vtab>() , 8usize);
-}
 impl Clone for sqlite3_vtab {
     fn clone(&self) -> Self { *self }
 }
@@ -1416,13 +1389,6 @@ pub struct sqlite3_index_info_sqlite3_index_constraint {
     pub usable: ::std::os::raw::c_uchar,
     pub iTermOffset: ::std::os::raw::c_int,
 }
-#[test]
-fn bindgen_test_layout_sqlite3_index_info_sqlite3_index_constraint() {
-    assert_eq!(::std::mem::size_of::<sqlite3_index_info_sqlite3_index_constraint>()
-               , 12usize);
-    assert_eq!(::std::mem::align_of::<sqlite3_index_info_sqlite3_index_constraint>()
-               , 4usize);
-}
 impl Clone for sqlite3_index_info_sqlite3_index_constraint {
     fn clone(&self) -> Self { *self }
 }
@@ -1432,13 +1398,6 @@ pub struct sqlite3_index_info_sqlite3_index_orderby {
     pub iColumn: ::std::os::raw::c_int,
     pub desc: ::std::os::raw::c_uchar,
 }
-#[test]
-fn bindgen_test_layout_sqlite3_index_info_sqlite3_index_orderby() {
-    assert_eq!(::std::mem::size_of::<sqlite3_index_info_sqlite3_index_orderby>()
-               , 8usize);
-    assert_eq!(::std::mem::align_of::<sqlite3_index_info_sqlite3_index_orderby>()
-               , 4usize);
-}
 impl Clone for sqlite3_index_info_sqlite3_index_orderby {
     fn clone(&self) -> Self { *self }
 }
@@ -1448,21 +1407,9 @@ pub struct sqlite3_index_info_sqlite3_index_constraint_usage {
     pub argvIndex: ::std::os::raw::c_int,
     pub omit: ::std::os::raw::c_uchar,
 }
-#[test]
-fn bindgen_test_layout_sqlite3_index_info_sqlite3_index_constraint_usage() {
-    assert_eq!(::std::mem::size_of::<sqlite3_index_info_sqlite3_index_constraint_usage>()
-               , 8usize);
-    assert_eq!(::std::mem::align_of::<sqlite3_index_info_sqlite3_index_constraint_usage>()
-               , 4usize);
-}
 impl Clone for sqlite3_index_info_sqlite3_index_constraint_usage {
     fn clone(&self) -> Self { *self }
 }
-#[test]
-fn bindgen_test_layout_sqlite3_index_info() {
-    assert_eq!(::std::mem::size_of::<sqlite3_index_info>() , 72usize);
-    assert_eq!(::std::mem::align_of::<sqlite3_index_info>() , 8usize);
-}
 impl Clone for sqlite3_index_info {
     fn clone(&self) -> Self { *self }
 }
@@ -1471,11 +1418,6 @@ impl Clone for sqlite3_index_info {
 pub struct sqlite3_vtab_cursor {
     pub pVtab: *mut sqlite3_vtab,
 }
-#[test]
-fn bindgen_test_layout_sqlite3_vtab_cursor() {
-    assert_eq!(::std::mem::size_of::<sqlite3_vtab_cursor>() , 8usize);
-    assert_eq!(::std::mem::align_of::<sqlite3_vtab_cursor>() , 8usize);
-}
 impl Clone for sqlite3_vtab_cursor {
     fn clone(&self) -> Self { *self }
 }
@@ -1600,11 +1542,6 @@ pub struct sqlite3_module {
                                                                 *const ::std::os::raw::c_char)
                                            -> ::std::os::raw::c_int>,
 }
-#[test]
-fn bindgen_test_layout_sqlite3_module() {
-    assert_eq!(::std::mem::size_of::<sqlite3_module>() , 160usize);
-    assert_eq!(::std::mem::align_of::<sqlite3_module>() , 8usize);
-}
 impl Clone for sqlite3_module {
     fn clone(&self) -> Self { *self }
 }
@@ -1727,11 +1664,6 @@ pub struct sqlite3_mutex_methods {
                                                                       *mut sqlite3_mutex)
                                                  -> ::std::os::raw::c_int>,
 }
-#[test]
-fn bindgen_test_layout_sqlite3_mutex_methods() {
-    assert_eq!(::std::mem::size_of::<sqlite3_mutex_methods>() , 72usize);
-    assert_eq!(::std::mem::align_of::<sqlite3_mutex_methods>() , 8usize);
-}
 impl Clone for sqlite3_mutex_methods {
     fn clone(&self) -> Self { *self }
 }
@@ -1829,11 +1761,6 @@ pub struct sqlite3_pcache_methods {
     pub xDestroy: ::std::option::Option<unsafe extern "C" fn(arg1:
                                                                  *mut sqlite3_pcache)>,
 }
-#[test]
-fn bindgen_test_layout_sqlite3_pcache_methods() {
-    assert_eq!(::std::mem::size_of::<sqlite3_pcache_methods>() , 88usize);
-    assert_eq!(::std::mem::align_of::<sqlite3_pcache_methods>() , 8usize);
-}
 impl Clone for sqlite3_pcache_methods {
     fn clone(&self) -> Self { *self }
 }
diff --git a/bindgen-bindings/bindgen_3.6.8.rs b/bindgen-bindings/bindgen_3.6.8.rs
index 3cd6b15..b72373b 100644
--- a/bindgen-bindings/bindgen_3.6.8.rs
+++ b/bindgen-bindings/bindgen_3.6.8.rs
@@ -321,21 +321,9 @@ pub struct sqlite3_file_sqlite3_io_methods {
                                                           ->
                                                               ::std::os::raw::c_int>,
 }
-#[test]
-fn bindgen_test_layout_sqlite3_file_sqlite3_io_methods() {
-    assert_eq!(::std::mem::size_of::<sqlite3_file_sqlite3_io_methods>() ,
-               104usize);
-    assert_eq!(::std::mem::align_of::<sqlite3_file_sqlite3_io_methods>() ,
-               8usize);
-}
 impl Clone for sqlite3_file_sqlite3_io_methods {
     fn clone(&self) -> Self { *self }
 }
-#[test]
-fn bindgen_test_layout_sqlite3_file() {
-    assert_eq!(::std::mem::size_of::<sqlite3_file>() , 8usize);
-    assert_eq!(::std::mem::align_of::<sqlite3_file>() , 8usize);
-}
 impl Clone for sqlite3_file {
     fn clone(&self) -> Self { *self }
 }
@@ -436,11 +424,6 @@ pub struct sqlite3_vfs {
                                                                       *mut ::std::os::raw::c_char)
                                                  -> ::std::os::raw::c_int>,
 }
-#[test]
-fn bindgen_test_layout_sqlite3_vfs() {
-    assert_eq!(::std::mem::size_of::<sqlite3_vfs>() , 136usize);
-    assert_eq!(::std::mem::align_of::<sqlite3_vfs>() , 8usize);
-}
 impl Clone for sqlite3_vfs {
     fn clone(&self) -> Self { *self }
 }
@@ -491,11 +474,6 @@ pub struct sqlite3_mem_methods {
                                                                   *mut ::std::os::raw::c_void)>,
     pub pAppData: *mut ::std::os::raw::c_void,
 }
-#[test]
-fn bindgen_test_layout_sqlite3_mem_methods() {
-    assert_eq!(::std::mem::size_of::<sqlite3_mem_methods>() , 64usize);
-    assert_eq!(::std::mem::align_of::<sqlite3_mem_methods>() , 8usize);
-}
 impl Clone for sqlite3_mem_methods {
     fn clone(&self) -> Self { *self }
 }
@@ -1368,11 +1346,6 @@ pub struct sqlite3_vtab {
     pub nRef: ::std::os::raw::c_int,
     pub zErrMsg: *mut ::std::os::raw::c_char,
 }
-#[test]
-fn bindgen_test_layout_sqlite3_vtab() {
-    assert_eq!(::std::mem::size_of::<sqlite3_vtab>() , 24usize);
-    assert_eq!(::std::mem::align_of::<sqlite3_vtab>() , 8usize);
-}
 impl Clone for sqlite3_vtab {
     fn clone(&self) -> Self { *self }
 }
@@ -1398,13 +1371,6 @@ pub struct sqlite3_index_info_sqlite3_index_constraint {
     pub usable: ::std::os::raw::c_uchar,
     pub iTermOffset: ::std::os::raw::c_int,
 }
-#[test]
-fn bindgen_test_layout_sqlite3_index_info_sqlite3_index_constraint() {
-    assert_eq!(::std::mem::size_of::<sqlite3_index_info_sqlite3_index_constraint>()
-               , 12usize);
-    assert_eq!(::std::mem::align_of::<sqlite3_index_info_sqlite3_index_constraint>()
-               , 4usize);
-}
 impl Clone for sqlite3_index_info_sqlite3_index_constraint {
     fn clone(&self) -> Self { *self }
 }
@@ -1414,13 +1380,6 @@ pub struct sqlite3_index_info_sqlite3_index_orderby {
     pub iColumn: ::std::os::raw::c_int,
     pub desc: ::std::os::raw::c_uchar,
 }
-#[test]
-fn bindgen_test_layout_sqlite3_index_info_sqlite3_index_orderby() {
-    assert_eq!(::std::mem::size_of::<sqlite3_index_info_sqlite3_index_orderby>()
-               , 8usize);
-    assert_eq!(::std::mem::align_of::<sqlite3_index_info_sqlite3_index_orderby>()
-               , 4usize);
-}
 impl Clone for sqlite3_index_info_sqlite3_index_orderby {
     fn clone(&self) -> Self { *self }
 }
@@ -1430,21 +1389,9 @@ pub struct sqlite3_index_info_sqlite3_index_constraint_usage {
     pub argvIndex: ::std::os::raw::c_int,
     pub omit: ::std::os::raw::c_uchar,
 }
-#[test]
-fn bindgen_test_layout_sqlite3_index_info_sqlite3_index_constraint_usage() {
-    assert_eq!(::std::mem::size_of::<sqlite3_index_info_sqlite3_index_constraint_usage>()
-               , 8usize);
-    assert_eq!(::std::mem::align_of::<sqlite3_index_info_sqlite3_index_constraint_usage>()
-               , 4usize);
-}
 impl Clone for sqlite3_index_info_sqlite3_index_constraint_usage {
     fn clone(&self) -> Self { *self }
 }
-#[test]
-fn bindgen_test_layout_sqlite3_index_info() {
-    assert_eq!(::std::mem::size_of::<sqlite3_index_info>() , 72usize);
-    assert_eq!(::std::mem::align_of::<sqlite3_index_info>() , 8usize);
-}
 impl Clone for sqlite3_index_info {
     fn clone(&self) -> Self { *self }
 }
@@ -1453,11 +1400,6 @@ impl Clone for sqlite3_index_info {
 pub struct sqlite3_vtab_cursor {
     pub pVtab: *mut sqlite3_vtab,
 }
-#[test]
-fn bindgen_test_layout_sqlite3_vtab_cursor() {
-    assert_eq!(::std::mem::size_of::<sqlite3_vtab_cursor>() , 8usize);
-    assert_eq!(::std::mem::align_of::<sqlite3_vtab_cursor>() , 8usize);
-}
 impl Clone for sqlite3_vtab_cursor {
     fn clone(&self) -> Self { *self }
 }
@@ -1582,11 +1524,6 @@ pub struct sqlite3_module {
                                                                 *const ::std::os::raw::c_char)
                                            -> ::std::os::raw::c_int>,
 }
-#[test]
-fn bindgen_test_layout_sqlite3_module() {
-    assert_eq!(::std::mem::size_of::<sqlite3_module>() , 160usize);
-    assert_eq!(::std::mem::align_of::<sqlite3_module>() , 8usize);
-}
 impl Clone for sqlite3_module {
     fn clone(&self) -> Self { *self }
 }
@@ -1709,11 +1646,6 @@ pub struct sqlite3_mutex_methods {
                                                                       *mut sqlite3_mutex)
                                                  -> ::std::os::raw::c_int>,
 }
-#[test]
-fn bindgen_test_layout_sqlite3_mutex_methods() {
-    assert_eq!(::std::mem::size_of::<sqlite3_mutex_methods>() , 72usize);
-    assert_eq!(::std::mem::align_of::<sqlite3_mutex_methods>() , 8usize);
-}
 impl Clone for sqlite3_mutex_methods {
     fn clone(&self) -> Self { *self }
 }
@@ -1811,11 +1743,6 @@ pub struct sqlite3_pcache_methods {
     pub xDestroy: ::std::option::Option<unsafe extern "C" fn(arg1:
                                                                  *mut sqlite3_pcache)>,
 }
-#[test]
-fn bindgen_test_layout_sqlite3_pcache_methods() {
-    assert_eq!(::std::mem::size_of::<sqlite3_pcache_methods>() , 88usize);
-    assert_eq!(::std::mem::align_of::<sqlite3_pcache_methods>() , 8usize);
-}
 impl Clone for sqlite3_pcache_methods {
     fn clone(&self) -> Self { *self }
 }
diff --git a/bindgen-bindings/bindgen_3.7.16.rs b/bindgen-bindings/bindgen_3.7.16.rs
index cd14895..557efdc 100644
--- a/bindgen-bindings/bindgen_3.7.16.rs
+++ b/bindgen-bindings/bindgen_3.7.16.rs
@@ -455,21 +455,9 @@ pub struct sqlite3_file_sqlite3_io_methods {
                                                                   ::std::os::raw::c_int)
                                              -> ::std::os::raw::c_int>,
 }
-#[test]
-fn bindgen_test_layout_sqlite3_file_sqlite3_io_methods() {
-    assert_eq!(::std::mem::size_of::<sqlite3_file_sqlite3_io_methods>() ,
-               136usize);
-    assert_eq!(::std::mem::align_of::<sqlite3_file_sqlite3_io_methods>() ,
-               8usize);
-}
 impl Clone for sqlite3_file_sqlite3_io_methods {
     fn clone(&self) -> Self { *self }
 }
-#[test]
-fn bindgen_test_layout_sqlite3_file() {
-    assert_eq!(::std::mem::size_of::<sqlite3_file>() , 8usize);
-    assert_eq!(::std::mem::align_of::<sqlite3_file>() , 8usize);
-}
 impl Clone for sqlite3_file {
     fn clone(&self) -> Self { *self }
 }
@@ -596,10 +584,6 @@ pub struct sqlite3_vfs {
                                                        *const ::std::os::raw::c_char>,
 }
 #[test]
-fn bindgen_test_layout_sqlite3_vfs() {
-    assert_eq!(::std::mem::size_of::<sqlite3_vfs>() , 168usize);
-    assert_eq!(::std::mem::align_of::<sqlite3_vfs>() , 8usize);
-}
 impl Clone for sqlite3_vfs {
     fn clone(&self) -> Self { *self }
 }
@@ -652,10 +636,6 @@ pub struct sqlite3_mem_methods {
     pub pAppData: *mut ::std::os::raw::c_void,
 }
 #[test]
-fn bindgen_test_layout_sqlite3_mem_methods() {
-    assert_eq!(::std::mem::size_of::<sqlite3_mem_methods>() , 64usize);
-    assert_eq!(::std::mem::align_of::<sqlite3_mem_methods>() , 8usize);
-}
 impl Clone for sqlite3_mem_methods {
     fn clone(&self) -> Self { *self }
 }
@@ -1604,10 +1584,6 @@ pub struct sqlite3_vtab {
     pub zErrMsg: *mut ::std::os::raw::c_char,
 }
 #[test]
-fn bindgen_test_layout_sqlite3_vtab() {
-    assert_eq!(::std::mem::size_of::<sqlite3_vtab>() , 24usize);
-    assert_eq!(::std::mem::align_of::<sqlite3_vtab>() , 8usize);
-}
 impl Clone for sqlite3_vtab {
     fn clone(&self) -> Self { *self }
 }
@@ -1633,13 +1609,6 @@ pub struct sqlite3_index_info_sqlite3_index_constraint {
     pub usable: ::std::os::raw::c_uchar,
     pub iTermOffset: ::std::os::raw::c_int,
 }
-#[test]
-fn bindgen_test_layout_sqlite3_index_info_sqlite3_index_constraint() {
-    assert_eq!(::std::mem::size_of::<sqlite3_index_info_sqlite3_index_constraint>()
-               , 12usize);
-    assert_eq!(::std::mem::align_of::<sqlite3_index_info_sqlite3_index_constraint>()
-               , 4usize);
-}
 impl Clone for sqlite3_index_info_sqlite3_index_constraint {
     fn clone(&self) -> Self { *self }
 }
@@ -1649,13 +1618,6 @@ pub struct sqlite3_index_info_sqlite3_index_orderby {
     pub iColumn: ::std::os::raw::c_int,
     pub desc: ::std::os::raw::c_uchar,
 }
-#[test]
-fn bindgen_test_layout_sqlite3_index_info_sqlite3_index_orderby() {
-    assert_eq!(::std::mem::size_of::<sqlite3_index_info_sqlite3_index_orderby>()
-               , 8usize);
-    assert_eq!(::std::mem::align_of::<sqlite3_index_info_sqlite3_index_orderby>()
-               , 4usize);
-}
 impl Clone for sqlite3_index_info_sqlite3_index_orderby {
     fn clone(&self) -> Self { *self }
 }
@@ -1665,21 +1627,9 @@ pub struct sqlite3_index_info_sqlite3_index_constraint_usage {
     pub argvIndex: ::std::os::raw::c_int,
     pub omit: ::std::os::raw::c_uchar,
 }
-#[test]
-fn bindgen_test_layout_sqlite3_index_info_sqlite3_index_constraint_usage() {
-    assert_eq!(::std::mem::size_of::<sqlite3_index_info_sqlite3_index_constraint_usage>()
-               , 8usize);
-    assert_eq!(::std::mem::align_of::<sqlite3_index_info_sqlite3_index_constraint_usage>()
-               , 4usize);
-}
 impl Clone for sqlite3_index_info_sqlite3_index_constraint_usage {
     fn clone(&self) -> Self { *self }
 }
-#[test]
-fn bindgen_test_layout_sqlite3_index_info() {
-    assert_eq!(::std::mem::size_of::<sqlite3_index_info>() , 72usize);
-    assert_eq!(::std::mem::align_of::<sqlite3_index_info>() , 8usize);
-}
 impl Clone for sqlite3_index_info {
     fn clone(&self) -> Self { *self }
 }
@@ -1688,11 +1638,6 @@ impl Clone for sqlite3_index_info {
 pub struct sqlite3_vtab_cursor {
     pub pVtab: *mut sqlite3_vtab,
 }
-#[test]
-fn bindgen_test_layout_sqlite3_vtab_cursor() {
-    assert_eq!(::std::mem::size_of::<sqlite3_vtab_cursor>() , 8usize);
-    assert_eq!(::std::mem::align_of::<sqlite3_vtab_cursor>() , 8usize);
-}
 impl Clone for sqlite3_vtab_cursor {
     fn clone(&self) -> Self { *self }
 }
@@ -1832,11 +1777,6 @@ pub struct sqlite3_module {
                                                                     ::std::os::raw::c_int)
                                                -> ::std::os::raw::c_int>,
 }
-#[test]
-fn bindgen_test_layout_sqlite3_module() {
-    assert_eq!(::std::mem::size_of::<sqlite3_module>() , 184usize);
-    assert_eq!(::std::mem::align_of::<sqlite3_module>() , 8usize);
-}
 impl Clone for sqlite3_module {
     fn clone(&self) -> Self { *self }
 }
@@ -1963,11 +1903,6 @@ pub struct sqlite3_mutex_methods {
                                                                       *mut sqlite3_mutex)
                                                  -> ::std::os::raw::c_int>,
 }
-#[test]
-fn bindgen_test_layout_sqlite3_mutex_methods() {
-    assert_eq!(::std::mem::size_of::<sqlite3_mutex_methods>() , 72usize);
-    assert_eq!(::std::mem::align_of::<sqlite3_mutex_methods>() , 8usize);
-}
 impl Clone for sqlite3_mutex_methods {
     fn clone(&self) -> Self { *self }
 }
@@ -2022,11 +1957,6 @@ pub struct sqlite3_pcache_page {
     pub pBuf: *mut ::std::os::raw::c_void,
     pub pExtra: *mut ::std::os::raw::c_void,
 }
-#[test]
-fn bindgen_test_layout_sqlite3_pcache_page() {
-    assert_eq!(::std::mem::size_of::<sqlite3_pcache_page>() , 16usize);
-    assert_eq!(::std::mem::align_of::<sqlite3_pcache_page>() , 8usize);
-}
 impl Clone for sqlite3_pcache_page {
     fn clone(&self) -> Self { *self }
 }
@@ -2084,11 +2014,6 @@ pub struct sqlite3_pcache_methods2 {
     pub xShrink: ::std::option::Option<unsafe extern "C" fn(arg1:
                                                                 *mut sqlite3_pcache)>,
 }
-#[test]
-fn bindgen_test_layout_sqlite3_pcache_methods2() {
-    assert_eq!(::std::mem::size_of::<sqlite3_pcache_methods2>() , 104usize);
-    assert_eq!(::std::mem::align_of::<sqlite3_pcache_methods2>() , 8usize);
-}
 impl Clone for sqlite3_pcache_methods2 {
     fn clone(&self) -> Self { *self }
 }
@@ -2141,11 +2066,6 @@ pub struct sqlite3_pcache_methods {
     pub xDestroy: ::std::option::Option<unsafe extern "C" fn(arg1:
                                                                  *mut sqlite3_pcache)>,
 }
-#[test]
-fn bindgen_test_layout_sqlite3_pcache_methods() {
-    assert_eq!(::std::mem::size_of::<sqlite3_pcache_methods>() , 88usize);
-    assert_eq!(::std::mem::align_of::<sqlite3_pcache_methods>() , 8usize);
-}
 impl Clone for sqlite3_pcache_methods {
     fn clone(&self) -> Self { *self }
 }
@@ -2254,11 +2174,6 @@ pub struct sqlite3_rtree_geometry {
     pub xDelUser: ::std::option::Option<unsafe extern "C" fn(arg1:
                                                                  *mut ::std::os::raw::c_void)>,
 }
-#[test]
-fn bindgen_test_layout_sqlite3_rtree_geometry() {
-    assert_eq!(::std::mem::size_of::<sqlite3_rtree_geometry>() , 40usize);
-    assert_eq!(::std::mem::align_of::<sqlite3_rtree_geometry>() , 8usize);
-}
 impl Clone for sqlite3_rtree_geometry {
     fn clone(&self) -> Self { *self }
 }
diff --git a/bindgen-bindings/bindgen_3.7.7.rs b/bindgen-bindings/bindgen_3.7.7.rs
index 37bebe3..7140eb8 100644
--- a/bindgen-bindings/bindgen_3.7.7.rs
+++ b/bindgen-bindings/bindgen_3.7.7.rs
@@ -421,21 +421,9 @@ pub struct sqlite3_file_sqlite3_io_methods {
                                                                   ::std::os::raw::c_int)
                                              -> ::std::os::raw::c_int>,
 }
-#[test]
-fn bindgen_test_layout_sqlite3_file_sqlite3_io_methods() {
-    assert_eq!(::std::mem::size_of::<sqlite3_file_sqlite3_io_methods>() ,
-               136usize);
-    assert_eq!(::std::mem::align_of::<sqlite3_file_sqlite3_io_methods>() ,
-               8usize);
-}
 impl Clone for sqlite3_file_sqlite3_io_methods {
     fn clone(&self) -> Self { *self }
 }
-#[test]
-fn bindgen_test_layout_sqlite3_file() {
-    assert_eq!(::std::mem::size_of::<sqlite3_file>() , 8usize);
-    assert_eq!(::std::mem::align_of::<sqlite3_file>() , 8usize);
-}
 impl Clone for sqlite3_file {
     fn clone(&self) -> Self { *self }
 }
@@ -561,11 +549,6 @@ pub struct sqlite3_vfs {
                                                    ->
                                                        *const ::std::os::raw::c_char>,
 }
-#[test]
-fn bindgen_test_layout_sqlite3_vfs() {
-    assert_eq!(::std::mem::size_of::<sqlite3_vfs>() , 168usize);
-    assert_eq!(::std::mem::align_of::<sqlite3_vfs>() , 8usize);
-}
 impl Clone for sqlite3_vfs {
     fn clone(&self) -> Self { *self }
 }
@@ -617,11 +600,6 @@ pub struct sqlite3_mem_methods {
                                                                   *mut ::std::os::raw::c_void)>,
     pub pAppData: *mut ::std::os::raw::c_void,
 }
-#[test]
-fn bindgen_test_layout_sqlite3_mem_methods() {
-    assert_eq!(::std::mem::size_of::<sqlite3_mem_methods>() , 64usize);
-    assert_eq!(::std::mem::align_of::<sqlite3_mem_methods>() , 8usize);
-}
 impl Clone for sqlite3_mem_methods {
     fn clone(&self) -> Self { *self }
 }
@@ -1532,11 +1510,6 @@ pub struct sqlite3_vtab {
     pub nRef: ::std::os::raw::c_int,
     pub zErrMsg: *mut ::std::os::raw::c_char,
 }
-#[test]
-fn bindgen_test_layout_sqlite3_vtab() {
-    assert_eq!(::std::mem::size_of::<sqlite3_vtab>() , 24usize);
-    assert_eq!(::std::mem::align_of::<sqlite3_vtab>() , 8usize);
-}
 impl Clone for sqlite3_vtab {
     fn clone(&self) -> Self { *self }
 }
@@ -1562,13 +1535,6 @@ pub struct sqlite3_index_info_sqlite3_index_constraint {
     pub usable: ::std::os::raw::c_uchar,
     pub iTermOffset: ::std::os::raw::c_int,
 }
-#[test]
-fn bindgen_test_layout_sqlite3_index_info_sqlite3_index_constraint() {
-    assert_eq!(::std::mem::size_of::<sqlite3_index_info_sqlite3_index_constraint>()
-               , 12usize);
-    assert_eq!(::std::mem::align_of::<sqlite3_index_info_sqlite3_index_constraint>()
-               , 4usize);
-}
 impl Clone for sqlite3_index_info_sqlite3_index_constraint {
     fn clone(&self) -> Self { *self }
 }
@@ -1578,13 +1544,6 @@ pub struct sqlite3_index_info_sqlite3_index_orderby {
     pub iColumn: ::std::os::raw::c_int,
     pub desc: ::std::os::raw::c_uchar,
 }
-#[test]
-fn bindgen_test_layout_sqlite3_index_info_sqlite3_index_orderby() {
-    assert_eq!(::std::mem::size_of::<sqlite3_index_info_sqlite3_index_orderby>()
-               , 8usize);
-    assert_eq!(::std::mem::align_of::<sqlite3_index_info_sqlite3_index_orderby>()
-               , 4usize);
-}
 impl Clone for sqlite3_index_info_sqlite3_index_orderby {
     fn clone(&self) -> Self { *self }
 }
@@ -1594,21 +1553,9 @@ pub struct sqlite3_index_info_sqlite3_index_constraint_usage {
     pub argvIndex: ::std::os::raw::c_int,
     pub omit: ::std::os::raw::c_uchar,
 }
-#[test]
-fn bindgen_test_layout_sqlite3_index_info_sqlite3_index_constraint_usage() {
-    assert_eq!(::std::mem::size_of::<sqlite3_index_info_sqlite3_index_constraint_usage>()
-               , 8usize);
-    assert_eq!(::std::mem::align_of::<sqlite3_index_info_sqlite3_index_constraint_usage>()
-               , 4usize);
-}
 impl Clone for sqlite3_index_info_sqlite3_index_constraint_usage {
     fn clone(&self) -> Self { *self }
 }
-#[test]
-fn bindgen_test_layout_sqlite3_index_info() {
-    assert_eq!(::std::mem::size_of::<sqlite3_index_info>() , 72usize);
-    assert_eq!(::std::mem::align_of::<sqlite3_index_info>() , 8usize);
-}
 impl Clone for sqlite3_index_info {
     fn clone(&self) -> Self { *self }
 }
@@ -1617,11 +1564,6 @@ impl Clone for sqlite3_index_info {
 pub struct sqlite3_vtab_cursor {
     pub pVtab: *mut sqlite3_vtab,
 }
-#[test]
-fn bindgen_test_layout_sqlite3_vtab_cursor() {
-    assert_eq!(::std::mem::size_of::<sqlite3_vtab_cursor>() , 8usize);
-    assert_eq!(::std::mem::align_of::<sqlite3_vtab_cursor>() , 8usize);
-}
 impl Clone for sqlite3_vtab_cursor {
     fn clone(&self) -> Self { *self }
 }
@@ -1761,11 +1703,6 @@ pub struct sqlite3_module {
                                                                     ::std::os::raw::c_int)
                                                -> ::std::os::raw::c_int>,
 }
-#[test]
-fn bindgen_test_layout_sqlite3_module() {
-    assert_eq!(::std::mem::size_of::<sqlite3_module>() , 184usize);
-    assert_eq!(::std::mem::align_of::<sqlite3_module>() , 8usize);
-}
 impl Clone for sqlite3_module {
     fn clone(&self) -> Self { *self }
 }
@@ -1892,11 +1829,6 @@ pub struct sqlite3_mutex_methods {
                                                                       *mut sqlite3_mutex)
                                                  -> ::std::os::raw::c_int>,
 }
-#[test]
-fn bindgen_test_layout_sqlite3_mutex_methods() {
-    assert_eq!(::std::mem::size_of::<sqlite3_mutex_methods>() , 72usize);
-    assert_eq!(::std::mem::align_of::<sqlite3_mutex_methods>() , 8usize);
-}
 impl Clone for sqlite3_mutex_methods {
     fn clone(&self) -> Self { *self }
 }
@@ -1994,11 +1926,6 @@ pub struct sqlite3_pcache_methods {
     pub xDestroy: ::std::option::Option<unsafe extern "C" fn(arg1:
                                                                  *mut sqlite3_pcache)>,
 }
-#[test]
-fn bindgen_test_layout_sqlite3_pcache_methods() {
-    assert_eq!(::std::mem::size_of::<sqlite3_pcache_methods>() , 88usize);
-    assert_eq!(::std::mem::align_of::<sqlite3_pcache_methods>() , 8usize);
-}
 impl Clone for sqlite3_pcache_methods {
     fn clone(&self) -> Self { *self }
 }
@@ -2102,11 +2029,6 @@ pub struct sqlite3_rtree_geometry {
     pub xDelUser: ::std::option::Option<unsafe extern "C" fn(arg1:
                                                                  *mut ::std::os::raw::c_void)>,
 }
-#[test]
-fn bindgen_test_layout_sqlite3_rtree_geometry() {
-    assert_eq!(::std::mem::size_of::<sqlite3_rtree_geometry>() , 40usize);
-    assert_eq!(::std::mem::align_of::<sqlite3_rtree_geometry>() , 8usize);
-}
 impl Clone for sqlite3_rtree_geometry {
     fn clone(&self) -> Self { *self }
 }
-- 
2.35.1

